lib/bdev/bdev.c

```diff
@@ -857,6 +859,38 @@ spdk_bdev_reset(struct spdk_bdev *bdev, enum spdk_bdev_reset_type reset_type,
 	return rc;
 }
 
+struct spdk_bdev_io *
+spdk_bdev_nvme_admin_passthru(struct spdk_bdev *bdev, const struct spdk_nvme_cmd *cmd,
+			      void *buf, size_t nbytes, spdk_bdev_io_completion_cb cb,
+			      void *cb_arg)
+{
+	struct spdk_bdev_io *bdev_io;
+	int rc;
+	uint32_t p_lcore = spdk_env_get_current_core();
+
+	bdev_io = spdk_bdev_get_io();
+	if (!bdev_io) {
+		SPDK_ERRLOG("bdev_io memory allocation failed during nvme_admin_passthru\n");
+		return NULL;
+	}
+
+	bdev_io->type = SPDK_BDEV_IO_TYPE_NVME_ADMIN;
+	memcpy(&bdev_io->u.nvme_admin_passthru.cmd, cmd, sizeof(struct spdk_nvme_cmd));
```

Comment:
This is better written as a direct assignment because then you can't get the size wrong.

bdev_io->u.nvme_admin_passthru.cmd = *cmd;